src/main/java/org/example/bookstore/validator/UniqueIsbnValidator.java

```diff
@@ -0,0 +1,22 @@
+package org.example.bookstore.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import lombok.RequiredArgsConstructor;
+import org.example.bookstore.service.BookService;
+
+@RequiredArgsConstructor
+public class UniqueIsbnValidator implements ConstraintValidator<UniqueIsbn, String> {
+
+    private final BookService bookService;
```

Comment:
read my comment about that validation once again, I said that we would want this validation in other service methods exactly to avoid that 😅 
Having services/repositories in the DTO validators is a bad practice, usually annotation-based validations are very lightweight and used for some basic things like formats, but anything that would need a db check is done in the service methods like `create`/`update`. Also this won't work because what if you would want to update a book and leave the same ISBN that was before - you'd have to send the same one but it would already exist in the DB, so you'll need to also check if it is different from the one that we already have in the `updateBook()` method